source/CommandLine/main.cpp

```diff
@@ -42,16 +42,20 @@ int VIREO_MAIN(int argc, const char * argv[])
     if (argc >= 2) {
         gShells._pRootShell = TypeManager::New(nullptr);
 
+        Boolean debugging = false;
         for (Int32 arg = 1; arg < argc; ++arg) {
             if (strcmp(argv[arg], "-D") == 0) {
                 gShells._pRootShell->DumpPrimitiveDictionary();
-                continue;
+            } else if (strcmp(argv[arg], "--debugging")) {
+                debugging = true;
             }
+        }
```

Comment:
I moved the processing of special command-line arguments into a separate loop so that we can detect the debugging flag and set it for all Via executions. I wanted to use a simple -D flag but that is already being used to print the primitives dictionary in use and hence went with a "--debugging" flag.